evalml/tests/automl_tests/test_automl.py

```diff
@@ -4474,3 +4688,69 @@ def test_automl_baseline_pipeline_predictions_and_scores_time_series(problem_typ
     np.testing.assert_allclose(
         baseline.feature_importance.iloc[:, 1], np.array([0.0] * X.shape[1])
     )
+
+
+@pytest.mark.parametrize(
+    "objective,errors",
+    [
+        ("Log Loss Binary", True),
+        ("AUC", True),
+        ("F1", False),
+        ("Accuracy Binary", False),
+    ],
+)
+@patch(
+    "evalml.objectives.binary_classification_objective.BinaryClassificationObjective.optimize_threshold",
+    return_value=0.65,
+)
+def test_automl_thresholding_objective(mock_optimize, objective, errors, X_y_binary):
+    X, y = X_y_binary
+    if errors:
+        with pytest.raises(
+            ValueError, match="Thresholding objective must be thresholdable"
+        ):
+            AutoMLSearch(
+                X_train=X,
+                y_train=y,
+                problem_type="binary",
+                thresholding_objective=objective,
+            )
+    else:
+        automl = AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            thresholding_objective=objective,
+            max_iterations=2,
+        )
+        automl.search()
```

Comment:
@bchen1116 how long does this test take to run?

Could you please mock `BinaryClassificationPipeline.fit` and `score` like we do in other tests? I bet that will reduce the runtime quite a bit, without changing the test behavior. Same for the other new test.

Reply:
It takes about 4 seconds to run. I can set max_iterations even lower and get it to run in under 2 seconds. This is comparable to the runtime when we mock, but with this, and the following test, I wanted to make sure the thresholds were being set properly. Would it still be preferable to mock the `fit`, `score`, `predict_proba`, and other methods here?